feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt

```diff
@@ -232,13 +232,19 @@ internal class AmazonBilling constructor(
         executeRequestOnUIThread { connectionError ->
             if (connectionError == null) {
                 purchaseHandler.purchase(
+                    activity,
                     appUserID,
                     storeProduct,
                     presentedOfferingIdentifier,
+                    onPurchaseCompleted = {
+                        purchaseHandler.onPurchaseCompleted(activity)
```

Comment:
This will hold a reference to the `activity` until this is done... There could be some issues if this never ended for any reason. One hacky possibility is to pass a `WeakReference` to the activity... Not very clean though...

Reply:
genuine question as an iOS dev used to weak references: what's the not-clean part?

Reply:
Well, in Java/Kotlin, it's not common to hold weak references since it's the norm to rely on the GC to free memory. I would say it's a technique only used when absolutely necessary and it's generally a better idea to rearchitect things as needed to make sure we don't hold anything that shouldn't be stored. I don't think there is a technical reason, and it's more of an architecture/style thing.

That said, in this case, it could cause actual issues since it's possible for the original activity to be freed and we never `finish` the `ProxyAmazonBillingActivity` so it could cause us to leave that orphaned activity as well.